Oddities.WinHelp/BTreeIndexHeader.cs

```diff
@@ -4,30 +4,39 @@
 namespace Oddities.WinHelp;
 
 [PublicAPI]
-public struct BTreeIndexHeader
+public readonly record struct BTreeIndexHeader
 {
-    public ushort Unused;
-    public short NEntries;
-    public short PreviousPage;
-    public short NextPage;
-    public DirectoryIndexEntry[] Entries;
+    public readonly ushort Unused;
+    public readonly short NEntries;
+    public readonly short PreviousPage;
+    public readonly short NextPage;
+    public readonly DirectoryIndexEntry[] Entries;
+
+    public BTreeIndexHeader(ushort unused, short nEntries, short previousPage, short nextPage, DirectoryIndexEntry[] entries)
+    {
+        Unused = unused;
+        NEntries = nEntries;
+        PreviousPage = previousPage;
+        NextPage = nextPage;
+        Entries = entries;
+    }
```

Comment:
This is good, but could be made better: use primary constructor. The whole struct will look like this then:
```
public record struct BTreeIndexHeader (
    ushort Unused,
    short NEntries,
    short PreviousPage,
    short NextPage,
    DirectoryIndexEntry[] Entries
)
```

Reply:
In this struct, I have to add `public` in front of `DirectoryIndexEntry[] Entries;`, otherwise I am getting error in WinHelp.cs  line  no 63

Reply:
I have tried to make changes as per your suggestion, but I could not understand it completely. Please review and provide feedback.

Reply:
I meant exactly this code:
```csharp
[PublicAPI]
public record struct BTreeIndexHeader(
    ushort Unused,
    short NEntries,
    short PreviousPage,
    short NextPage,
    DirectoryIndexEntry[] Entries
)
{
    public static BTreeIndexHeader Load(BinaryReader data, Encoding fileNameEncoding)
    {
        var unused = data.ReadUInt16();
        var nEntries = data.ReadInt16();
        var previousPage = data.ReadInt16();
        var nextPage = data.ReadInt16();
        var entries = new DirectoryIndexEntry[nEntries];
        
        for (var i = 0; i < nEntries; ++i)
            entries[i] = DirectoryIndexEntry.Load(data, fileNameEncoding);

        return new BTreeIndexHeader(unused, nEntries, previousPage, nextPage, entries);
    }
}
```

Could you try to do that, please? In your commits, I see that you've tried some other changes (with curly braces instead of parens), and that's probably why you were having problems with them.

There's no need for `public`, because the fields in the primary constructor declaration are already automatically public.